flytekit/extras/sqlalchemy/task.py

```diff
@@ -0,0 +1,84 @@
+import os
+import typing
+from dataclasses import dataclass
+
+import pandas as pd
+from sqlalchemy import create_engine
+
+from flytekit import current_context, FlyteContext, kwtypes
+from flytekit.core.base_sql_task import SQLTask
+from flytekit.core.python_function_task import PythonInstanceTask
+from flytekit.types.schema import FlyteSchema
+
+
+@dataclass
+class SQLAlchemyConfig(object):
+    """
+    Use this configuration to configure task. String should be standard
+    sqlalchemy connector format
+    (https://docs.sqlalchemy.org/en/14/core/engines.html#database-urls).
+    Database can be found:
+      - within the container
+      - or from a publicly accessible source
+
+    Args:
+        uri: default sqlalchemy connector
+        connect_args: sqlalchemy kwarg overrides -- ex: host
+        secret_connect_args: flyte secrets loaded into sqlalchemy connect args
+            -- ex: {"password": {"name": SECRET_NAME, "group": SECRET_GROUP}}
+    """
+
+    uri: str
+    connect_args: typing.Optional[typing.Dict[str, typing.Any]] = None
+    secret_connect_args: typing.Optional[typing.Dict[str, typing.Dict[str, typing.Any]]] = None
+
+
+class SQLAlchemyTask(PythonInstanceTask[SQLAlchemyConfig], SQLTask[SQLAlchemyConfig]):
+    """
+    Makes it possible to run client side SQLAlchemy queries that optionally return a FlyteSchema object
+
+    TODO: How should we use pre-built containers for running portable tasks like this. Should this always be a
+          referenced task type?
+    """
+
+    _SQLALCHEMY_TASK_TYPE = "sqlalchemy"
+
+    def __init__(
+        self,
+        name: str,
+        query_template: str,
+        task_config: SQLAlchemyConfig,
+        inputs: typing.Optional[typing.Dict[str, typing.Type]] = None,
+        output_schema_type: typing.Optional[typing.Type[FlyteSchema]] = None,
+        **kwargs,
+    ):
+        outputs = kwtypes(results=output_schema_type if output_schema_type else FlyteSchema)
+        self._uri = task_config.uri
+        self._connect_args = task_config.connect_args or {}
+        if task_config.secret_connect_args is not None:
+            for key, secret in task_config.secret_connect_args.items():
+                if "name" in secret and "group" in secret:
+                    value = current_context().secrets.get(secret["group"], secret["name"])
```

Comment:
you can only access the secrets in the `execute` command or logically. They will be available here, but you should be using them only in the execute. Reason is __init__ will be invoked at convert / compile time, but execute only at runtime.

Also you will need to pass the secret requests to thparent 
https://github.com/flyteorg/flytekit/blob/master/flytekit/core/python_auto_container.py#L47

super().__init__(..., secret_requests=[...])

and access them in execute

Reply:
got it. the secret_requests should pass through in kwargs